vector/src/main/java/im/vector/app/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -429,6 +440,61 @@ class MessageActionsViewModel @AssistedInject constructor(@Assisted
         }
     }
 
+    /**
+     * Determine whether or not the Reply In Thread bottom sheet setting will be visible
+     * to the user
+     */
+    private fun canReplyInThread(event: TimelineEvent,
+                                 messageContent: MessageContent?,
+                                 actionPermissions: ActionPermissions): Boolean {
+        // Only event of type EventType.MESSAGE are supported for the moment
+        if (!BuildConfig.THREADING_ENABLED) return false
+        if (initialState.isFromThreadTimeline) return false
+        if (event.root.getClearType() != EventType.MESSAGE &&
+                !event.isSticker() && !event.isPoll()) return false
+        if (!actionPermissions.canSendMessage) return false
+        return when (messageContent?.msgType) {
+            MessageType.MSGTYPE_TEXT,
+            MessageType.MSGTYPE_NOTICE,
+            MessageType.MSGTYPE_EMOTE,
+            MessageType.MSGTYPE_IMAGE,
+            MessageType.MSGTYPE_VIDEO,
+            MessageType.MSGTYPE_AUDIO,
+            MessageType.MSGTYPE_FILE,
+            MessageType.MSGTYPE_POLL_START,
+            MessageType.MSGTYPE_STICKER_LOCAL -> true
+            else                              -> false
+        }
+    }
+
+    /**
+     * Determine whether or no the selected event is a root thread event from within
+     * a thread timeline
+     */
+    private fun canViewInRoom(event: TimelineEvent,
+                              messageContent: MessageContent?,
+                              actionPermissions: ActionPermissions): Boolean {
+        // Only event of type EventType.MESSAGE are supported for the moment
+        if (!BuildConfig.THREADING_ENABLED) return false
+        if (!initialState.isFromThreadTimeline) return false
```

Comment:
I see, thanks. But the other test below are a bit weird to me. We should test if this is the first event of the RecyclerView, no?

Reply:
Thanks I agree, we can remove some of the tests, I will try to minimise them. However, checking only if it's the first cell is not enough while this might change in a future feature. Maybe insert a header or something prior to that so might be better to be decoupled.